scripts/distribution/macOS-package/template/scripts/postinstall

```diff
@@ -17,6 +17,27 @@ function addBinariesToPath() {
     ln -f -s "/Library/Concordium Node/node-collector" "$symlinkCollector"
 }
 
+# Replaces all occurrences of 'placeholder' with 'replacement' in file 'fileName'.
+# The replacement can contain any characters except for special characters in XML, which must be escaped.
+function replaceInFile() {
+    local placeholder=$1
+    local replacement=$2
+    local fileName=$3
+
+    # Other tools, such as sed and awk, has limitations to the characters allowed, so '/' and '&' could not be used.
```

Comment:
Wouldn't sed work?
```
~ echo "//foo//" > foo.txt
~ sed -i'' -e 's/\//1/g' foo.txt
~ cat foo.txt                   
~ 11foo11
```

Reply:
No, because the `1` in this case is provided by the user and so it might contain a `/` or some other character which has special meaning in sed.

Reply:
Surely you can just escape them?

Reply:
```
~ echo "xxxfooxxx" > foo.txt     
~ sed -i'' -e 's/x/\//g' foo.txt
~ cat foo.txt                    
~///foo///
```
Or what am I missing? My main objection we're dragging a python dependency into the build for something that seems to be solvable with just sed? But I might be missing your point... :-)

Reply:
My recommendation is to just whitelist some characters in the name, say [a-zA-Z0-9.-, ] and check this in the installer. when the user is entering them. Then we don't have to worry about escaping anything. If they really want a fancy name with smileys they can edit the configuration file manually.